openhands/runtime/client/runtime.py

```diff
@@ -34,6 +36,62 @@
 from openhands.runtime.utils.runtime_build import build_runtime_image
 
 
+class LogBuffer:
+    """
+    Synchronous buffer for Docker container logs.
+
+    This class provides a thread-safe way to collect, store, and retrieve logs
+    from a Docker container. It uses a deque with a maximum length to store log
+    lines and provides methods for appending, retrieving, and clearing logs.
+    """
+
+    def __init__(self, container: docker.models.containers.Container, maxlen=10000):
+        self.buffer: deque[str] = deque(maxlen=maxlen)
+        self.lock = threading.Lock()
+        self.log_generator = container.logs(stream=True, follow=True)
+        self.log_stream_thread = threading.Thread(target=self.stream_logs)
+        self.log_stream_thread.daemon = True
+        self.log_stream_thread.start()
+        self._stop_event = threading.Event()
+
+    def append(self, log_line: str):
+        with self.lock:
+            self.buffer.append(log_line)
+
+    def get_and_clear(self) -> list[str]:
+        with self.lock:
+            logs = list(self.buffer)
+            self.buffer.clear()
+            return logs
```

Comment:
We not directly make `self.buffer` as a list[str].

Reply:
do you mean "why not?"